v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/RangeCountDoFn.java

```diff
@@ -97,43 +103,49 @@ public void processElement(@Element Range input, OutputReceiver<Range> out, Proc
               this.countQuery, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
       stmt.setQueryTimeout((int) ((this.timeoutMillis + TIMEOUT_GRACE_MILLIS) / 1000));
       new RangePreparedStatementSetter(numColumns).setParameters(input, stmt);
-      try {
-        ResultSet rs = stmt.executeQuery();
-        if (rs.next()) {
-          count = rs.getLong(1);
-          if (rs.wasNull()) {
-            logger.error(
-                "Got null for count resultSet. Range = {}, Query = {}, DataSource = {}",
-                input,
-                countQuery,
-                dataSource);
-            count = Range.INDETERMINATE_COUNT;
-          }
-        } else {
+      ResultSet rs = stmt.executeQuery();
+      if (rs.next()) {
+        count = rs.getLong(1);
+        if (rs.wasNull()) {
           logger.error(
-              "Got empty count resultSet. Range = {}, Query = {}, DataSource = {}",
+              "Got null for count resultSet. Range = {}, Query = {}, DataSource = {}",
               input,
               countQuery,
               dataSource);
           count = Range.INDETERMINATE_COUNT;
         }
-      } catch (SQLTimeoutException e) {
+      } else {
+        logger.error(
+            "Got empty count resultSet. Range = {}, Query = {}, DataSource = {}",
+            input,
+            countQuery,
+            dataSource);
+        count = Range.INDETERMINATE_COUNT;
```

Comment:
if we get an empty, then this should be set to 0. Only in the case of exception it makes sense to go back to an undetermined number.

Reply:
Ideally count(*) query should never return empty, it does return 0 for no rows.
making the count `Range.INDETERMINATE_COUNT` marks the range for recounting.

This case is mostly from a code completeness (or utilize return value) sake, we need to have some handle for rs.next() returning false, since it's unexpected, making it `Range.INDETERMINATE_COUNT` will mark the range for recounting.

We have error logs for this to catch it if we hit this in a real world situation.